File: Examples/FxSocket.Sftp/Windows/CS/SftpDesktopClientDemo/VerifierForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace FxSocketSamples
{
	/// <summary>
	/// Summary description for Verifier.
	/// </summary>
	public class VerifierForm : System.Windows.Forms.Form
	{
		/// <summary>
		/// Required designer variable.
		/// </summary>
        private System.ComponentModel.Container components = null;
		private System.Windows.Forms.Button btnReject;
        private System.Windows.Forms.Button btnAccept;
		private System.Windows.Forms.Button btnOkAndTrust;

		private bool _accepted = false;
        private Label lblHostname;
        private Label label8;
        private Label label6;
        private Label lblValidTo;
        private Label lblValidFrom;
        private Label lblIssuer;
        private Label lblSubject;
        private Label lblProblem;
        private Panel panel2;
        private Label label5;
        private Label label4;
        private Label label3;
        private Label label2;
        private Label label1;
		private bool _addIssuerCertificateAuthothorityToTrustedCaStore = false;

		public string Hostname
		{
			set { lblHostname.Text = value; }
		}

		public string Subject
		{
			set { lblSubject.Text = value; }
		}

		public string Issuer
		{
			set { lblIssuer.Text = value; }
		}

		public string ValidFrom
		{
			set { lblValidFrom.Text = value; }
		}

		public string ValidTo
		{
			set { lblValidTo.Text = value; }
		}

		public string Problem
		{
			set { lblProblem.Text = value; }
		}

		public bool Accepted
		{
			get { return _accepted; }
		}

		public bool AddIssuerCertificateAuthothorityToTrustedCaStore
		{
			get { return _addIssuerCertificateAuthothorityToTrustedCaStore; }
		}

		public bool ShowAddIssuerToTrustedCaStoreButton
		{
			get { return btnOkAndTrust.Visible; }
			set { btnOkAndTrust.Visible = value; }
		}

		public VerifierForm()
		{
			InitializeComponent();
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose(bool disposing)
		{
			if (disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(disposing);
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.btnReject = new System.Windows.Forms.Button();
            this.btnAccept = new System.Windows.Forms.Button();
            this.btnOkAndTrust = new System.Windows.Forms.Button();
            this.label1 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.label3 = new System.Windows.Forms.Label();
            this.label4 = new System.Windows.Forms.Label();
            this.label5 = new System.Windows.Forms.Label();
            this.panel2 = new System.Windows.Forms.Panel();
            this.lblProblem = new System.Windows.Forms.Label();
            this.lblSubject = new System.Windows.Forms.Label();
            this.lblIssuer = new System.Windows.Forms.Label();
            this.lblValidFrom = new System.Windows.Forms.Label();
            this.lblValidTo = new System.Windows.Forms.Label();
            this.label6 = new System.Windows.Forms.Label();
            this.label8 = new System.Windows.Forms.Label();
            this.lblHostname = new System.Windows.Forms.Label();
            this.SuspendLayout();
            // 
            // btnReject
            // 
            this.btnReject.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.btnReject.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.btnReject.Location = new System.Drawing.Point(336, 329);
            this.btnReject.Name = "btnReject";
            this.btnReject.Size = new System.Drawing.Size(72, 23);
            this.btnReject.TabIndex = 2;
            this.btnReject.Text = "Reject";
            this.btnReject.Click += new System.EventHandler(this.btnReject_Click);
            // 
            // btnAccept
            // 
            this.btnAccept.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.btnAccept.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.btnAccept.Location = new System.Drawing.Point(99, 329);
            this.btnAccept.Name = "btnAccept";
            this.btnAccept.Size = new System.Drawing.Size(72, 23);
            this.btnAccept.TabIndex = 1;
            this.btnAccept.Text = "Accept";
            this.btnAccept.Click += new System.EventHandler(this.btnAccept_Click);
            // 
            // btnOkAndTrust
            // 
            this.btnOkAndTrust.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.btnOkAndTrust.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.btnOkAndTrust.Location = new System.Drawing.Point(177, 329);
            this.btnOkAndTrust.Name = "btnOkAndTrust";
            this.btnOkAndTrust.Size = new System.Drawing.Size(153, 23);
            this.btnOkAndTrust.TabIndex = 5;
            this.btnOkAndTrust.Text = "Always Trust this Authority";
            this.btnOkAndTrust.Visible = false;
            this.btnOkAndTrust.Click += new System.EventHandler(this.btnOkAndTrust_Click);
            // 
            // label1
            // 
            this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(238)));
            this.label1.Location = new System.Drawing.Point(8, 8);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(256, 23);
            this.label1.TabIndex = 0;
            this.label1.Text = "CERTIFICATE INFORMATION:";
            // 
            // label2
            // 
            this.label2.Location = new System.Drawing.Point(8, 56);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(72, 23);
            this.label2.TabIndex = 1;
            this.label2.Text = "Subject:";
            // 
            // label3
            // 
            this.label3.Location = new System.Drawing.Point(8, 104);
            this.label3.Name = "label3";
            this.label3.Size = new System.Drawing.Size(72, 23);
            this.label3.TabIndex = 2;
            this.label3.Text = "Issuer:";
            // 
            // label4
            // 
            this.label4.Location = new System.Drawing.Point(8, 152);
            this.label4.Name = "label4";
            this.label4.Size = new System.Drawing.Size(72, 23);
            this.label4.TabIndex = 3;
            this.label4.Text = "Valid from:";
            // 
            // label5
            // 
            this.label5.Location = new System.Drawing.Point(8, 176);
            this.label5.Name = "label5";
            this.label5.Size = new System.Drawing.Size(72, 23);
            this.label5.TabIndex = 4;
            this.label5.Text = "Valid to:";
            // 
            // panel2
            // 
            this.panel2.BackColor = System.Drawing.SystemColors.ControlDarkDark;
            this.panel2.Location = new System.Drawing.Point(8, 200);
            this.panel2.Name = "panel2";
            this.panel2.Size = new System.Drawing.Size(384, 3);
            this.panel2.TabIndex = 5;
            // 
            // lblProblem
            // 
            this.lblProblem.ForeColor = System.Drawing.Color.Red;
            this.lblProblem.Location = new System.Drawing.Point(8, 216);
            this.lblProblem.Name = "lblProblem";
            this.lblProblem.Size = new System.Drawing.Size(384, 72);
            this.lblProblem.TabIndex = 6;
            // 
            // lblSubject
            // 
            this.lblSubject.Location = new System.Drawing.Point(80, 56);
            this.lblSubject.Name = "lblSubject";
            this.lblSubject.Size = new System.Drawing.Size(312, 40);
            this.lblSubject.TabIndex = 7;
            // 
            // lblIssuer
            // 
            this.lblIssuer.Location = new System.Drawing.Point(80, 104);
            this.lblIssuer.Name = "lblIssuer";
            this.lblIssuer.Size = new System.Drawing.Size(312, 40);
            this.lblIssuer.TabIndex = 8;
            // 
            // lblValidFrom
            // 
            this.lblValidFrom.Location = new System.Drawing.Point(80, 152);
            this.lblValidFrom.Name = "lblValidFrom";
            this.lblValidFrom.Size = new System.Drawing.Size(224, 23);
            this.lblValidFrom.TabIndex = 9;
            // 
            // lblValidTo
            // 
            this.lblValidTo.Location = new System.Drawing.Point(80, 176);
            this.lblValidTo.Name = "lblValidTo";
            this.lblValidTo.Size = new System.Drawing.Size(224, 23);
            this.lblValidTo.TabIndex = 10;
            // 
            // label6
            // 
            this.label6.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(238)));
            this.label6.Location = new System.Drawing.Point(8, 8);
            this.label6.Name = "label6";
            this.label6.Size = new System.Drawing.Size(216, 23);
            this.label6.TabIndex = 11;
            this.label6.Text = "Certificate details:";
            // 
            // label8
            // 
            this.label8.Location = new System.Drawing.Point(8, 32);
            this.label8.Name = "label8";
            this.label8.Size = new System.Drawing.Size(72, 23);
            this.label8.TabIndex = 12;
            this.label8.Text = "Hostname:";
            // 
            // lblHostname
            // 
            this.lblHostname.Location = new System.Drawing.Point(80, 32);
            this.lblHostname.Name = "lblHostname";
            this.lblHostname.Size = new System.Drawing.Size(312, 16);
            this.lblHostname.TabIndex = 13;
            // 
            // VerifierForm
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(418, 364);
            this.ControlBox = false;
            this.Controls.Add(this.btnOkAndTrust);
            this.Controls.Add(this.btnAccept);
            this.Controls.Add(this.btnReject);
            this.Controls.Add(this.lblHostname);
            this.Controls.Add(this.label8);
            this.Controls.Add(this.label6);
            this.Controls.Add(this.lblValidTo);
            this.Controls.Add(this.lblValidFrom);
            this.Controls.Add(this.lblIssuer);
            this.Controls.Add(this.lblSubject);
            this.Controls.Add(this.lblProblem);
            this.Controls.Add(this.panel2);
            this.Controls.Add(this.label5);
            this.Controls.Add(this.label4);
            this.Controls.Add(this.label3);
            this.Controls.Add(this.label2);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.MaximizeBox = false;
            this.Name = "VerifierForm";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Certificate";
            this.ResumeLayout(false);

		}
		#endregion

		private void btnAccept_Click(object sender, System.EventArgs e)
		{
			_accepted = true;
			this.Close();
		}

		private void btnReject_Click(object sender, System.EventArgs e)
		{
			this.Close();
		}

		private void btnOkAndTrust_Click(object sender, System.EventArgs e)
		{
			_accepted = true;
			_addIssuerCertificateAuthothorityToTrustedCaStore = true;

			this.Close();
		}

	}
}

File: Examples/FxSocket.Sftp/Windows/VB/SftpDesktopClientDemo/VerifierForm.vb
Imports System.Collections
Imports System.ComponentModel

Namespace FxSocketSamples
	''' <summary>
	''' Summary description for Verifier.
	''' </summary>
	Public Class VerifierForm
		Inherits System.Windows.Forms.Form
		''' <summary>
		''' Required designer variable.
		''' </summary>
		Private components As System.ComponentModel.Container = Nothing
		Private WithEvents btnReject As System.Windows.Forms.Button
		Private WithEvents btnAccept As System.Windows.Forms.Button
		Private WithEvents btnOkAndTrust As System.Windows.Forms.Button

		Private _accepted As Boolean = False
		Private lblHostname As Label
		Private label8 As Label
		Private label6 As Label
		Private lblValidTo As Label
		Private lblValidFrom As Label
		Private lblIssuer As Label
		Private lblSubject As Label
		Private lblProblem As Label
		Private panel2 As Panel
		Private label5 As Label
		Private label4 As Label
		Private label3 As Label
		Private label2 As Label
		Private label1 As Label
		Private _addIssuerCertificateAuthothorityToTrustedCaStore As Boolean = False

		Public WriteOnly Property Hostname() As String
			Set(ByVal value As String)
				lblHostname.Text = value
			End Set
		End Property

		Public WriteOnly Property Subject() As String
			Set(ByVal value As String)
				lblSubject.Text = value
			End Set
		End Property

		Public WriteOnly Property Issuer() As String
			Set(ByVal value As String)
				lblIssuer.Text = value
			End Set
		End Property

		Public WriteOnly Property ValidFrom() As String
			Set(ByVal value As String)
				lblValidFrom.Text = value
			End Set
		End Property

		Public WriteOnly Property ValidTo() As String
			Set(ByVal value As String)
				lblValidTo.Text = value
			End Set
		End Property

		Public WriteOnly Property Problem() As String
			Set(ByVal value As String)
				lblProblem.Text = value
			End Set
		End Property

		Public ReadOnly Property Accepted() As Boolean
			Get
				Return _accepted
			End Get
		End Property

		Public ReadOnly Property AddIssuerCertificateAuthothorityToTrustedCaStore() As Boolean
			Get
				Return _addIssuerCertificateAuthothorityToTrustedCaStore
			End Get
		End Property

		Public Property ShowAddIssuerToTrustedCaStoreButton() As Boolean
			Get
				Return btnOkAndTrust.Visible
			End Get
			Set(ByVal value As Boolean)
				btnOkAndTrust.Visible = value
			End Set
		End Property

		Public Sub New()
			InitializeComponent()
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If components IsNot Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.btnReject = New System.Windows.Forms.Button()
			Me.btnAccept = New System.Windows.Forms.Button()
			Me.btnOkAndTrust = New System.Windows.Forms.Button()
			Me.label1 = New System.Windows.Forms.Label()
			Me.label2 = New System.Windows.Forms.Label()
			Me.label3 = New System.Windows.Forms.Label()
			Me.label4 = New System.Windows.Forms.Label()
			Me.label5 = New System.Windows.Forms.Label()
			Me.panel2 = New System.Windows.Forms.Panel()
			Me.lblProblem = New System.Windows.Forms.Label()
			Me.lblSubject = New System.Windows.Forms.Label()
			Me.lblIssuer = New System.Windows.Forms.Label()
			Me.lblValidFrom = New System.Windows.Forms.Label()
			Me.lblValidTo = New System.Windows.Forms.Label()
			Me.label6 = New System.Windows.Forms.Label()
			Me.label8 = New System.Windows.Forms.Label()
			Me.lblHostname = New System.Windows.Forms.Label()
			Me.SuspendLayout()
			' 
			' btnReject
			' 
			Me.btnReject.Anchor = (CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.btnReject.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.btnReject.Location = New System.Drawing.Point(336, 329)
			Me.btnReject.Name = "btnReject"
			Me.btnReject.Size = New System.Drawing.Size(72, 23)
			Me.btnReject.TabIndex = 2
			Me.btnReject.Text = "Reject"
'			Me.btnReject.Click += New System.EventHandler(Me.btnReject_Click)
			' 
			' btnAccept
			' 
			Me.btnAccept.Anchor = (CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.btnAccept.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.btnAccept.Location = New System.Drawing.Point(99, 329)
			Me.btnAccept.Name = "btnAccept"
			Me.btnAccept.Size = New System.Drawing.Size(72, 23)
			Me.btnAccept.TabIndex = 1
			Me.btnAccept.Text = "Accept"
'			Me.btnAccept.Click += New System.EventHandler(Me.btnAccept_Click)
			' 
			' btnOkAndTrust
			' 
			Me.btnOkAndTrust.Anchor = (CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.btnOkAndTrust.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.btnOkAndTrust.Location = New System.Drawing.Point(177, 329)
			Me.btnOkAndTrust.Name = "btnOkAndTrust"
			Me.btnOkAndTrust.Size = New System.Drawing.Size(153, 23)
			Me.btnOkAndTrust.TabIndex = 5
			Me.btnOkAndTrust.Text = "Always Trust this Authority"
			Me.btnOkAndTrust.Visible = False
'			Me.btnOkAndTrust.Click += New System.EventHandler(Me.btnOkAndTrust_Click)
			' 
			' label1
			' 
			Me.label1.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, (CByte(238)))
			Me.label1.Location = New System.Drawing.Point(8, 8)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(256, 23)
			Me.label1.TabIndex = 0
			Me.label1.Text = "CERTIFICATE INFORMATION:"
			' 
			' label2
			' 
			Me.label2.Location = New System.Drawing.Point(8, 56)
			Me.label2.Name = "label2"
			Me.label2.Size = New System.Drawing.Size(72, 23)
			Me.label2.TabIndex = 1
			Me.label2.Text = "Subject:"
			' 
			' label3
			' 
			Me.label3.Location = New System.Drawing.Point(8, 104)
			Me.label3.Name = "label3"
			Me.label3.Size = New System.Drawing.Size(72, 23)
			Me.label3.TabIndex = 2
			Me.label3.Text = "Issuer:"
			' 
			' label4
			' 
			Me.label4.Location = New System.Drawing.Point(8, 152)
			Me.label4.Name = "label4"
			Me.label4.Size = New System.Drawing.Size(72, 23)
			Me.label4.TabIndex = 3
			Me.label4.Text = "Valid from:"
			' 
			' label5
			' 
			Me.label5.Location = New System.Drawing.Point(8, 176)
			Me.label5.Name = "label5"
			Me.label5.Size = New System.Drawing.Size(72, 23)
			Me.label5.TabIndex = 4
			Me.label5.Text = "Valid to:"
			' 
			' panel2
			' 
			Me.panel2.BackColor = System.Drawing.SystemColors.ControlDarkDark
			Me.panel2.Location = New System.Drawing.Point(8, 200)
			Me.panel2.Name = "panel2"
			Me.panel2.Size = New System.Drawing.Size(384, 3)
			Me.panel2.TabIndex = 5
			' 
			' lblProblem
			' 
			Me.lblProblem.ForeColor = System.Drawing.Color.Red
			Me.lblProblem.Location = New System.Drawing.Point(8, 216)
			Me.lblProblem.Name = "lblProblem"
			Me.lblProblem.Size = New System.Drawing.Size(384, 72)
			Me.lblProblem.TabIndex = 6
			' 
			' lblSubject
			' 
			Me.lblSubject.Location = New System.Drawing.Point(80, 56)
			Me.lblSubject.Name = "lblSubject"
			Me.lblSubject.Size = New System.Drawing.Size(312, 40)
			Me.lblSubject.TabIndex = 7
			' 
			' lblIssuer
			' 
			Me.lblIssuer.Location = New System.Drawing.Point(80, 104)
			Me.lblIssuer.Name = "lblIssuer"
			Me.lblIssuer.Size = New System.Drawing.Size(312, 40)
			Me.lblIssuer.TabIndex = 8
			' 
			' lblValidFrom
			' 
			Me.lblValidFrom.Location = New System.Drawing.Point(80, 152)
			Me.lblValidFrom.Name = "lblValidFrom"
			Me.lblValidFrom.Size = New System.Drawing.Size(224, 23)
			Me.lblValidFrom.TabIndex = 9
			' 
			' lblValidTo
			' 
			Me.lblValidTo.Location = New System.Drawing.Point(80, 176)
			Me.lblValidTo.Name = "lblValidTo"
			Me.lblValidTo.Size = New System.Drawing.Size(224, 23)
			Me.lblValidTo.TabIndex = 10
			' 
			' label6
			' 
			Me.label6.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, (CByte(238)))
			Me.label6.Location = New System.Drawing.Point(8, 8)
			Me.label6.Name = "label6"
			Me.label6.Size = New System.Drawing.Size(216, 23)
			Me.label6.TabIndex = 11
			Me.label6.Text = "Certificate details:"
			' 
			' label8
			' 
			Me.label8.Location = New System.Drawing.Point(8, 32)
			Me.label8.Name = "label8"
			Me.label8.Size = New System.Drawing.Size(72, 23)
			Me.label8.TabIndex = 12
			Me.label8.Text = "Hostname:"
			' 
			' lblHostname
			' 
			Me.lblHostname.Location = New System.Drawing.Point(80, 32)
			Me.lblHostname.Name = "lblHostname"
			Me.lblHostname.Size = New System.Drawing.Size(312, 16)
			Me.lblHostname.TabIndex = 13
			' 
			' VerifierForm
			' 
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
			Me.ClientSize = New System.Drawing.Size(418, 364)
			Me.ControlBox = False
			Me.Controls.Add(Me.btnOkAndTrust)
			Me.Controls.Add(Me.btnAccept)
			Me.Controls.Add(Me.btnReject)
			Me.Controls.Add(Me.lblHostname)
			Me.Controls.Add(Me.label8)
			Me.Controls.Add(Me.label6)
			Me.Controls.Add(Me.lblValidTo)
			Me.Controls.Add(Me.lblValidFrom)
			Me.Controls.Add(Me.lblIssuer)
			Me.Controls.Add(Me.lblSubject)
			Me.Controls.Add(Me.lblProblem)
			Me.Controls.Add(Me.panel2)
			Me.Controls.Add(Me.label5)
			Me.Controls.Add(Me.label4)
			Me.Controls.Add(Me.label3)
			Me.Controls.Add(Me.label2)
			Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
			Me.MaximizeBox = False
			Me.Name = "VerifierForm"
			Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
			Me.Text = "Certificate"
			Me.ResumeLayout(False)

		End Sub
		#End Region

		Private Sub btnAccept_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnAccept.Click
			_accepted = True
			Me.Close()
		End Sub

		Private Sub btnReject_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnReject.Click
			Me.Close()
		End Sub

		Private Sub btnOkAndTrust_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnOkAndTrust.Click
			_accepted = True
			_addIssuerCertificateAuthothorityToTrustedCaStore = True

			Me.Close()
		End Sub

	End Class
End Namespace
